### PR TITLE
feat: analytical Eagle3 BlockMask builder (O(num_blocks) memory)

### PR DESCRIPTION
**Pull request by yubofredwang**
## Summary

- Add `build_eagle3_block_mask`, an O(num_blocks)-memory direct constructor for the Eagle3 causal+suffix `BlockMask`. `create_block_mask` materialises the full `(Q_LEN, KV_LEN)` boolean grid internally (~112 GB at Q=49K, KV=245K), causing OOM on long-context Eagle3 training. The analytical builder writes the sparse `kv_indices`/`q_indices` tensors directly from the known mask structure, dropping peak memory to a few MB.
- Add `eagle3_block_mask` dispatcher: uses the analytical builder when the preconditions hold (`Q_LEN % BLOCK_SIZE == 0`, `KV_LEN % BLOCK_SIZE == 0`, `KV_LEN % Q_LEN == 0`), otherwise falls back to `create_block_mask` / `compile_friendly_create_block_mask` (covers tests/edge cases where the O(Q*KV) cost is irrelevant).
- Simplify `generate_eagle3_mask` to drop the `seq_lengths`-aware padding mask. It's now only consumed by the dispatcher's fallback path; padding is enforced upstream via the attention mask, and the analytical path gets exact equivalence by construction (verified by tests below).
- Re-export `build_eagle3_block_mask` and `eagle3_block_mask` from `torchspec.models.ops`.

## Testing

New file: `tests/test_build_eagle3_block_mask.py` (18 tests + 13 subtests):

Local run on torch 2.9.1+cu129, single CUDA device:

```
$ pytest tests/test_build_eagle3_block_mask.py -x -q
..................                                          [100%]
18 passed, 1 warning, 13 subtests passed in 7.52s
```

Smoke import check confirms `LlamaFlexAttention` and `DeepSeekMLAFlexAttention` still load with the dispatcher import. No new lints.

Solves https://github.com/lightseekorg/TorchSpec/issues/86

### CODE REVIEW

File: torchspec/models/ops/flex_attention.py [hunk 1]
```diff
@@ -116,19 +117,171 @@ def compile_friendly_create_block_mask(
...
-        padding_mask = (kv_idx < seq_lengths[b]) & (q_idx < seq_lengths[b])
-        return causal_mask & padding_mask
+        return q_idx >= kv_idx
 
     def suffix_mask(b, h, q_idx, kv_idx):
-        suffix_mask = kv_idx >= Q_LEN
-        padding_mask = kv_idx % Q_LEN < seq_lengths[b]
-        diagnol_mask = (kv_idx - q_idx) % Q_LEN == 0
-        return suffix_mask & padding_mask & diagnol_mask
+        return (kv_idx >= Q_LEN) & ((kv_idx - q_idx) % Q_LEN == 0)
```
[the remark this answers is not in the document]
Comment (yubofredwang) [reply torchspec/models/ops/flex_attention.py:133]: This is useful, but we have not supported padded batch yet. so... until we support that, this is not a concern
